spec.md

```diff
@@ -2374,6 +2407,38 @@ SInt("h-2A")
 SInt("h-2a")
 ```
 
+## Enum Expressions
+
+An enumeration can be constructed by specifying the variant tag, a data value
+expression, and a return type specifier. The data value expression may be
+omitted when the data type is `UInt<0>(0)`, where it is implicitly defined to
+be `UInt<0>(0)`.
+
+``` firrtl
+#a : [#a, #b, #c]
```

Comment:
What about the same syntax as UInt construction?  `[#some : UInt<8>, #none](#some UInt<8>(1))`